### ISSUE TITLE
Old bug - Heidi sql exception when executing queries

### ISSUE THREAD

**Issue by ianpmurphy** [HeidiSQL/HeidiSQL#413]
## Steps to reproduce this issue
1. Step 1;
Leave heidi open for a hours/a day or so

2. Step 2;
(I think the issue is related to my accessing my machine remotely via RDP and then returning - i.e the screen is resized)
Execute a query

3. Step N;
4. Then I get...
Exception report

## Current behavior
Heidi generates an exception dialog and offers the possibility to save the report.

## Expected behavior

no exception

## Possible solution

No idea

## Environment

Win 10 x64 es-ES

* **HeidiSQL version:** 
I am currently running 9.5.0.5264 but this issue has been around for years. I had it under win 7 and it still occurs with win 10. I've only just recently started collecting the bugreports

[heidi-bugreport-2018-11-12.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624156/heidi-bugreport-2018-11-12.txt)
[Heidi-bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624157/Heidi-bugreport.txt)
[Heidi-bugreport2.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624158/Heidi-bugreport2.txt)
[heidi-bugreport3.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624159/heidi-bugreport3.txt)
[heidi-bugreport4.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624160/heidi-bugreport4.txt)
[heidi-bugreport5.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624161/heidi-bugreport5.txt)
[heidi-bugreport6.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624162/heidi-bugreport6.txt)
[heidi-bugreport7.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624163/heidi-bugreport7.txt)
[heidi-bugreport8.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624164/heidi-bugreport8.txt)
[heidi-bugreport9.txt](https://github.com/HeidiSQL/HeidiSQL/files/2624165/heidi-bugreport9.txt)

**Comment by ansgarbecker** [comment]
All your crash reports are based on old versions. Could you please at least post one report you made with the current buld.

**Comment by ianpmurphy** [comment]
I will update, though I am running 9.5 on a pretty recent version, so unless you have specifically fixed this bug the issue should still apply. I've seen this issue with every version I've ever run and I've been using it for several years, so its not like its something which has turned up in the latest release.

As it occurs pretty much every day I will try and provoke it. I am fairly sure it has something to do with resizing as I do a lot of remote work and rdp into my machine. The issue seems to occur after working remotely. Having said that, I also rarely reboot, so it could also be a memory leak or something else related to being open for a long time.

**Comment by taxpayer** [comment]
Hi @ianpmurphy,

I used to have a problem similar to this one you're reporting and I could solve it setting the "Ping every X second" option in the database registration screen, as shown below. Before it was zero, after I've set it to 10 secs, the "coming back to Heidi after some time not using it" problem has gone.

Maybe should you give it a try?

![image](https://user-images.githubusercontent.com/9285710/49688245-369da500-faf6-11e8-8855-21ebff3aa607.png)

**Comment by ansgarbecker** [comment]
@ianpmurphy , please give feedback whether the issue was solved in the latest build. I guess so, as i recently fixed #57

**Comment by ianpmurphy** [comment]
Hi Ansgar,
Hopefully I can reopen this. Sorry for the delay but I did as you asked and updated to the most recent version I could pull down - 9.5.0.5362. This version has been way more stable, which is kinda annoying as I actually wanted to hit the problem.
Its taken a while but after a few weeks of occasional usage, it finally generated an exception when I hit F9.
I am attaching the bugreport.
[bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/2723643/bugreport.txt)

@taxpayer I tried your suggestion and it may have contributed to my connection being more stable, but I'm not sure if it was the nightly build or the option. In any case its been much better since. I think I'm going to try setting this back to 0 and see if it becomes more unstable again.
